src/Servers/HttpSys/src/FeatureContext.cs

```diff
@@ -600,20 +722,24 @@ internal Task OnCompleted()
             return NotifyOnCompletedAsync();
         }
 
-        private async Task NotifyOnCompletedAsync()
+        private Task NotifyOnCompletedAsync()
         {
             var actions = _onCompletedActions;
-            _onCompletedActions = null;
-            if (actions == null)
+            if (actions.Count == 0)
             {
-                return;
+                return Task.CompletedTask;
             }
 
-            actions.Reverse();
-            // Execute last to first. This mimics a stack unwind.
-            foreach (var actionPair in actions)
+            return Awaited(actions);
+
+            static async Task Awaited(List<(Func<object, Task> callback, object state)> actions)
```

Comment:
Nit: since this is duplicated in the same class, you might as well make it a method.

Reply:
Done